travis.sh

```diff
@@ -294,6 +285,10 @@ function prepare_ros_workspace() {
 
    # Validate that we have some packages to build
    test -z "$(colcon list)" && echo -e "$(colorize RED Workspace $ROS_WS has no packages to build. Terminating.)" && exit 1
+
+   # Source the UNDERLAY workspace
+   travis_run_simple --title "Sourcing the underlay workspace" source "${ROS_UNDERLAY:-/opt/ros/$ROS_DISTRO}"
```

Comment:
We need to source `setup.bash` and we should run it before calling `rosdep install` otherwise it will fail for not being able to find the packages in the underlay

Reply:
Also, it's complaining about `unbound variable` when sourcing the `setup.bash` I'm not sure why this's happening only in Travis

Reply:
Ok. I thought you disabled that for a potentially undefined `${ROS_UNDERLAY}`.